lib/modules/datasource/npm/yarnrc.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { loadConfigFromYarnrcYml, resolveRegistryUrl } from './yarnrc';
+
+describe('modules/datasource/npm/yarnrc', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('resolveRegistryUrl()', () => {
+    it('considers default registry', () => {
+      const registryUrl = resolveRegistryUrl('a-package', {
+        npmRegistryServer: 'https://private.example.com/npm',
+      });
+      expect(registryUrl).toBe('https://private.example.com/npm');
+    });
+
+    it('chooses matching scoped registry over default registry', () => {
+      const registryUrl = resolveRegistryUrl('@scope/a-package', {
+        npmRegistryServer: 'https://private.example.com/npm',
+        npmScopes: {
+          scope: {
+            npmRegistryServer: 'https://scope.example.com/npm',
+          },
+        },
+      });
+      expect(registryUrl).toBe('https://scope.example.com/npm');
+    });
+
+    it('ignores non matching scoped registry', () => {
+      const registryUrl = resolveRegistryUrl('@scope/a-package', {
+        npmScopes: {
+          'other-scope': {
+            npmRegistryServer: 'https://other-scope.example.com/npm',
+          },
+        },
+      });
+      expect(registryUrl).toBeNull();
+    });
+  });
+
+  describe('loadConfigFromYarnrcYml()', () => {
+    it.each([
```

Comment:
can we use the table syntax, so this becomes more readable?

Reply:
Unfortunately not, because we have multiple strings.

I moved the test data to fixtures instead. Let me know if that works for you!

Reply:
no, fixture files makes it even worse (IO perf)

I thought it's posible to format like this:
https://github.com/renovatebot/renovate/blob/6c69ba0cde562b560fcd38ddcf2ffdeaa80ea00d/lib/modules/datasource/metadata.spec.ts#L81-L99

Reply:
Unfortunately the table syntax does not work for multi-line values in the cells.

We could move the test inputs to variables though

Reply:
The table syntax does not really help here IMO. I moved back to the original approach and shortened the examples a bit. I think readability is OK-ish now.